File: C++/1.cpp
class Solution {
//     hashMap = {}
//         for i in range (0, len(nums)):
            
//             diff = target - nums[i]
            
//             if (diff in hashMap):
//                 return [hashMap[diff], i]
//             hashMap[nums[i]] = i 
//         return
public:
    vector<int> twoSum(vector<int>& nums, int target) {
        unordered_map<int, int> hashMap;
        for(int i = 0; i < nums.size(); i++){
            int diff = target - nums[i];
            if(hashMap.find(diff) != hashMap.end())
                return {hashMap[diff], i};
            hashMap[nums[i]] = i;
        }
        return {};
    }
};

File: JobPrep2024/1.py
class Solution(object):
    def twoSum(self, nums, target):
        """
        :type nums: List[int]
        :type target: int
        :rtype: List[int]
        """
        twoSumMap = {}
        for i in range(0, len(nums)):
            leftOver = target - nums[i]
            if leftOver in twoSumMap.keys():
                return [twoSumMap[leftOver], i]  
            else:
               twoSumMap[nums[i]] = i

        return []       
